pontoon/contributors/utils.py

```diff
@@ -478,39 +488,70 @@ def get_contribution_timeline_data(user, contribution_type=None, day=None):
+                total_count += info["count"]
+                # Count number of unique project/locale pairs
+                projects.add(key)
+
+            # Generate title for the localizations belonging to the same contribution type
+            p_count = len(projects)
+            if contribution_type == "user_translations":
+                title = f"Submitted { intcomma(total_count) } translation{ pluralize(total_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
+            elif contribution_type == "user_reviews":
+                title = f"Reviewed { intcomma(total_count) } suggestion{ pluralize(total_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
+            elif contribution_type == "peer_reviews":
+                title = f"Received review for { intcomma(total_count) } suggestion{ pluralize(total_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
+
+            # Add title as a field
+            if month not in contribution_data:
+                contribution_data[month] = {}
+            contribution_data[month]["title"] = title
+
+        # Generate localization URL and add it to the data dict
         params = params_map[contribution_type]
-        for key, val in contribution_data.items():
-            url = reverse(
-                "pontoon.translate",
-                args=[val["locale"]["code"], val["project"]["slug"], "all-resources"],
-            )
-            contribution_data[key]["url"] = f"{url}?{urlencode(params)}"
-
-        # Generate title for the localizations belonging to the same contribution type
-        if contribution_type == "user_translations":
-            title = f"Submitted { intcomma(c_count) } translation{ pluralize(c_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
-        elif contribution_type == "user_reviews":
-            title = f"Reviewed { intcomma(c_count) } suggestion{ pluralize(c_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
-        elif contribution_type == "peer_reviews":
-            title = f"Received review for { intcomma(c_count) } suggestion{ pluralize(c_count) } in { intcomma(p_count) } project{ pluralize(p_count) }"
+        for key, value in contribution_data.items():
+            for field, data in value.items():
+                if field != "title":
+                    url = reverse(
+                        "pontoon.translate",
+                        args=[
+                            data["locale"]["code"],
+                            data["project"]["slug"],
+                            "all-resources",
+                        ],
+                    )
+                    data["url"] = f"{url}?{urlencode(params)}"
 
         contributions.update(
             {
-                title: {
+                contribution_type: {
                     "data": contribution_data,
                     "type": contribution_type.replace("_", "-"),
                 }
             }
         )
```

Comment:
We should directly structure the `contributions` dict the way we need data to be structured. We should be able to do that without the additional `restructured` dict.